src/views/resultsView/createScanProvider.ts

```diff
@@ -53,6 +53,14 @@ async function createScanForProject(
   logs: Logs
 ) {
   const scanBranch: Item = context.workspaceState.get(constants.branchIdKey);
+  if(scanBranch.id === constants.localBranch){
+    const gitApi = await getGitAPIRepository();
+    const gitBranchName = gitApi.repositories[0]?.state.HEAD?.name;//TODO: replace with getFromState(context, constants.branchName) when the onBranchChange is working properly
```

Comment:
Also here:
//TODO: replace with getFromState(context, constants.branchName) when the onBranchChange is working properly

Reply:
No, we need to wait for the branch change listener.